frontend/src/components/AppHeader.vue

```diff
@@ -144,9 +158,23 @@ watch([useRoute(), articles], ([newRoute]) => {
     article.value = articlesStore.getArticleById(articleId as string);
   } else {
     article.value = null;
+    focusModeStore.disableFocusMode();
+    focusModeToggle.value = false;
   }
 });
 
+watch(isFocusMode, (newValue) => {
+  focusModeToggle.value = newValue;
+});
+
+function handleFocusModeToggle(value: boolean) {
+  if (value) {
+    focusModeStore.enableFocusMode();
+  } else {
+    focusModeStore.disableFocusMode();
+  }
```

Comment:
you can refactor this:

```suggestion
  focusModeStore.toggleFocusMode(value) 
```